Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4800,6 +4792,32 @@ - (void)extendRangeToIncludeUnobservedItems
     self.previousLastTimestamp = nil;
 }
 
+- (void)createNewMessageMappings
+{
+    if (self.thread.uniqueId.length > 0) {
+        self.messageMappings = [[YapDatabaseViewMappings alloc] initWithGroups:@[ self.thread.uniqueId ]
+                                                                          view:TSMessageDatabaseViewExtensionName];
+    } else {
+        OWSFail(@"uniqueId unexpectedly empty for thread: %@", self.thread);
```

Comment:
huh - is this ever known to happen?

Reply:
There was an "extremely hard to repro" bug of this form 1+ years ago.  I haven't seen it in the last year, but I know I wasn't the only person who saw it then.